lib/entities/quote/RelayQuote.ts

```diff
@@ -0,0 +1,304 @@
+      permitData: this.getPermitData(),
+      classicQuoteData: this.classicQuoteData,
+    };
+  }
+
+  public toOrder(): RelayOrder {
+    const orderBuilder = new RelayOrderBuilder(this.chainId);
+    const feeStartTime = Math.floor(Date.now() / 1000);
+    const deadline = feeStartTime + this.auctionPeriodSecs + this.deadlineBufferSecs;
+    const nonce = this.nonce ?? generateRandomNonce();
+
+    const builder = orderBuilder
+      .swapper(ethers.utils.getAddress(this.request.config.swapper))
+      .nonce(BigNumber.from(nonce))
+      .deadline(deadline)
+      // Add the swap input to UR
+      .input({
+        token: this.tokenIn,
+        amount: this.amountIn,
+        recipient: UNIVERSAL_ROUTER_ADDRESS(this.chainId),
+      })
+      // Add the gas token input to the filler
+      .fee({
+        token: this.request.config.gasToken,
+        startAmount: this.feeAmountStart,
+        endAmount: this.feeAmountEnd,
+        startTime: feeStartTime,
+        endTime: feeStartTime + this.auctionPeriodSecs,
+      })
+      .universalRouterCalldata(this.universalRouterCalldata(deadline));
+
+    return builder.build();
+  }
+
+  public toLog(): LogJSON {
+    return {
+      ...this.classicQuote.toLog(),
+      requestId: this.requestId,
+      quoteId: this.quoteId,
+      amountIn: this.amountIn.toString(),
+      amountOut: this.amountOut.toString(),
+      gasToken: this.request.config.gasToken,
+      feeAmountStart: this.feeAmountStart.toString(),
+      feeAmountEnd: this.feeAmountEnd.toString(),
+      swapper: this.swapper,
+      routing: RoutingType[this.routingType],
+      slippage: parseFloat(this.request.info.slippageTolerance),
+      createdAt: this.createdAt,
+      createdAtMs: this.createdAtMs,
+    };
+  }
+
+  getPermitData(): PermitBatchTransferFromData {
+    return this.toOrder().permitData();
+  }
+
+  public get slippage(): Percent {
+    return new Percent(parseFloat(this.request.info.slippageTolerance) * 100, 10_000);
+  }
+
+  public universalRouterCalldata(deadline: number): string {
+    return SwapRouter.swapCallParameters(
+      new UniswapTrade(this.classicQuote.toRouterTrade(), {
+        slippageTolerance: this.slippage,
+        recipient: this.swapper,
+        useRouterBalance: true,
+        deadlineOrPreviousBlockhash: deadline,
+      })
+    ).calldata;
+  }
+
+  // The number of seconds from now that fee escalation should begin
+  public get startTimeBufferSecs(): number {
+    if (this.request.config.startTimeBufferSecs !== undefined) {
+      return this.request.config.startTimeBufferSecs;
+    }
+
+    return DEFAULT_START_TIME_BUFFER_SECS;
+  }
+
+  // The number of seconds from startTime that fee escalation should end
+  public get auctionPeriodSecs(): number {
+    if (this.request.config.auctionPeriodSecs !== undefined) {
+      return this.request.config.auctionPeriodSecs;
+    }
+
+    switch (this.chainId) {
+      case 1:
+        return 60;
```

Comment:
We can add all these numbers as constants somewhere so they're easy to find and modify.

Reply:
Good suggestion, I won't go through and edit the values for DutchV2 since that's custom logic and will likely be tweakedi n the future